engine/controller/projection/controller.go

```diff
@@ -105,6 +124,31 @@ func (c *Controller) Handle(
 		return nil, nil
 	}
 
+	compactResult := make(chan error, 1)
```

Comment:
This channel is buffered because the error is expected to be written there before it is read from the channel, right?

Reply:
Yeah, in particular because we call into user code (the handler itself) before reading from the channel, and that code can panic (and infact *must* panic in some cases). 

If the channel wasn't buffered the compaction goroutine would hang "forever". Obviously this isn't really going to happen if the whole application panics, but who knows, maybe we recover from that panic in the future. I like to just generally cover my bases like this.

Looking at it now, something I haven't handled is what happens if `Compact()` itself panics. I will add some handling for this too.

Reply:
> Looking at it now, something I haven't handled is what happens if `Compact()` itself panics. I will add some handling for this too.

Ok this is a dumb suggestion on my part - we make this goroutine right there, there's no way someone else can recover from it!

Reply:
So I guess no recovery in `Compact()` then.